packages/core/admin/admin/src/content-manager/pages/ListConfiguration/ListConfigurationPage.tsx

```diff
@@ -0,0 +1,208 @@
+import * as React from 'react';
+
+import { ContentLayout, Divider, Flex, Layout, Main } from '@strapi/design-system';
+import {
+  CheckPagePermissions,
+  LoadingIndicatorPage,
+  useAPIErrorHandler,
+  useNotification,
+  useTracking,
+} from '@strapi/helper-plugin';
+import { useIntl } from 'react-intl';
+import { Navigate } from 'react-router-dom';
+
+import { useTypedSelector } from '../../../core/store/hooks';
+import { useEnterprise } from '../../../hooks/useEnterprise';
+import { Form, FormProps } from '../../components/Form';
+import { useDoc } from '../../hooks/useDocument';
+import { ListFieldLayout, ListLayout, useDocLayout } from '../../hooks/useDocumentLayout';
+import { useUpdateContentTypeConfigurationMutation } from '../../services/contentTypes';
+import { setIn } from '../../utils/object';
+
+import { Header } from './components/Header';
+import { Settings, SettingsProps } from './components/Settings';
+import { SortDisplayedFields } from './components/SortDisplayedFields';
+
+import type { Contracts } from '@strapi/plugin-content-manager/_internal/shared';
+
+const EXCLUDED_SORT_ATTRIBUTE_TYPES = [
+  'media',
+  'richtext',
+  'dynamiczone',
+  'relation',
+  'component',
+  'json',
+  'blocks',
+];
+
+interface FormData extends Pick<ListLayout, 'settings'> {
+  layout: Array<Pick<ListFieldLayout, 'sortable' | 'name'> & { label: string }>;
+}
+
+const ListConfiguration = () => {
+  const { formatMessage } = useIntl();
+  const { trackUsage } = useTracking();
+  const toggleNotification = useNotification();
+  const { _unstableFormatAPIError: formatAPIError } = useAPIErrorHandler();
+
+  const { model, collectionType, schema } = useDoc();
+
+  const { isLoading: isLoadingLayout, list, edit } = useDocLayout();
+
+  const [updateContentTypeConfiguration] = useUpdateContentTypeConfigurationMutation();
+  const handleSubmit: FormProps<FormData>['onSubmit'] = async (data, event) => {
+    event.preventDefault();
+
+    try {
+      trackUsage('willSaveContentTypeLayout');
+
+      /**
+       * We reconstruct the metadatas object by taking the existing edit metadatas
+       * and re-merging that by attribute name with the current list metadatas, whilst overwriting
+       * the data from the form we've built.
+       */
+      const meta = Object.entries(edit.metadatas).reduce<Contracts.ContentTypes.Metadatas>(
+        (acc, [name, editMeta]) => {
+          const { mainField: _mainField, ...listMeta } = list.metadatas[name];
+
+          const { label, sortable } = data.layout.find((field) => field.name === name) ?? {};
+
+          acc[name] = {
+            edit: editMeta,
+            list: {
+              ...listMeta,
+              label,
+              sortable,
+            },
```

Comment:
I don't have all the context, maybe this is something you are going to change on the listView 😅  but when doing this, the meta object ended up with label only on the fields that are selected to show up on the list table. 

That's fine for the table, but we use also these labels when showing the filters. So this is causing my app to crash.

See on my screenshot: On address content type I selected on the configure the view page to show only the "id" and the "city" fields, so in the payload all the other fields (for example "cover") doesn't have a label on the list meta, this cause the list view to crash when showing the Filters.

![Screenshot 2024-02-07 at 16 23 41](https://github.com/strapi/strapi/assets/23050694/d3ca7e84-b194-4d12-91ac-77659d02710e)

Reply:
I added validation to the field form because the label is required 👍🏼